SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -22,62 +23,109 @@ SpatialLoadBalanceEnforcer::SpatialLoadBalanceEnforcer(const PhysicalWorkerName&
 void SpatialLoadBalanceEnforcer::OnAuthorityIntentComponentUpdated(const Worker_ComponentUpdateOp& Op)
 {
 	check(Op.update.component_id == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID);
-	if (StaticComponentView->HasAuthority(Op.entity_id, SpatialConstants::ENTITY_ACL_COMPONENT_ID))
+
+	MaybeQueueAclAssignmentRequest(Op.entity_id);
```

Comment:
Would it make more sense to check here if we can actually care about this entity. Then we can call it EnsureAclAssignmentQueued and only deal with duplicated in the function. Then we can hopefully get rid of the awful comment that says where the function is called from.

Reply:
I'm still not entirely sure why we need to queue it. Shouldn't it just be a on-change-make-sure-two-things-match affair?

Reply:
I would rather it happened immediately, however I also think it is out of scope. @jennifer-at-improbable-io who I spoke to about this before mentioned it because so many requests might be made in the course of a tick that it is worth queueing and deduping them.

Reply:
This is a latency sensitive operation so I don't think we should be aggregating stuff more than needed unless there is data to back that up. Especially as almost all code complexity is because this queue is here.

Reply:
Resolving this as I think we've decided it's not part of this PR though.